Frameworks/CoreGraphics/CGFont.mm

```diff
@@ -16,615 +16,129 @@
+                                              "CGFont",
+                                              __CGFontInit, // init
+                                              NULL, // copy
+                                              __CGFontDeallocate, // deallocate
+                                              __CGFontEqual,
+                                              __CGFontHash,
+                                              NULL, //
+                                              __CGFontCopyDescription };
 
-    CGFontSetFTFontSize(font, face, size);
-
-    FT_Pos penX = 0;
-    FT_Pos penY = 0;
-
-    //  Lookup each glyph
-    for (i = 0; i < count; i++) {
-        error = FT_Load_Glyph(face, glyphs[i], FT_LOAD_NO_HINTING);
-        if (error != 0) {
-            TraceError(TAG, L"CGFontMeasureGlyphs: unknown char %x err=%d", glyphs[i], error);
-        } else {
-            /* increment pen position */
-            penX += slot->advance.x;
-            penY += slot->advance.y;
-        }
-    }
-
-    sizeOut->width = ((float)penX) / 64.0f;
-    sizeOut->height = ((float)face->size->metrics.ascender - face->size->metrics.descender) * spacing / 64.0f;
-
-    _CGFontUnlock();
-
-    return count;
-}
-
-DWORD CGFontGetGlyphs(id font, WORD* str, DWORD length, WORD* glyphs) {
-    DWORD i;
-    int glyphsOut = 0;
-
-    _CGFontLock();
-
-    //  Get the font
-    FT_Face face = (FT_Face)(DWORD)[font _sizingFontHandle];
-
-    //  Lookup each glyph
-    for (i = 0; i < length; i++) {
-        glyphs[glyphsOut] = FT_Get_Char_Index(face, str[i]);
-        if (glyphs[glyphsOut] == 0)
-            continue;
-        glyphsOut++;
-    }
-
-    _CGFontUnlock();
-
-    return glyphsOut;
-}
-
-/**
- @Status Interoperable
-*/
-CFDataRef CGFontCopyTableForTag(CGFontRef font, uint32_t tag) {
-    //  Get the font
-    FT_Face face = (FT_Face)(DWORD)[font _sizingFontHandle];
-
-    _CGFontLock();
-    FT_Error err;
-
-    DWORD numBytes = 0;
-    err = FT_Load_Sfnt_Table(face, tag, 0, NULL, &numBytes);
-
-    char* pData = (char*)IwMalloc(numBytes);
-    err = FT_Load_Sfnt_Table(face, tag, 0, (FT_Byte*)pData, &numBytes);
-
-    id ret = [[NSData alloc] initWithBytes:pData length:numBytes];
-    IwFree(pData);
-
-    _CGFontUnlock();
-
-    return (CFDataRef)ret;
-}
-
-/**
- @Status Caveat
- @Notes Returns family name
-*/
-CFStringRef CGFontCopyFullName(CGFontRef font) {
-    //  Get the font
-    FT_Face face = (FT_Face)(DWORD)[font _sizingFontHandle];
-
-    if ((face == NULL) || (face->family_name == NULL)) {
-        return nullptr;
-    }
-
-    return (CFStringRef)[[NSString stringWithCString:face->family_name] retain];
-}
-
-/**
- @Status Interoperable
-*/
-CGFontRef CGFontCreateWithDataProvider(CGDataProviderRef cgDataProvider) {
-    CFRetain(cgDataProvider);
-    return (CGFontRef)[[_LazyUIFont fontWithData:(NSData*)cgDataProvider] retain];
-}
+static CFTypeID __kCGFontTypeID = _CFRuntimeRegisterClass(&__CGFontClass);
```

Comment:
I think we should avoid registering these things on library load, _but_:
if you move this entire thing into CGFontGetTypeID, it will automatically only happen once, thread-safely, the first time somebody uses CGFont.